### PR TITLE
Concept Entry for Drag and Drop SwiftUI

### PR DESCRIPTION
**Pull request by arirays**
### Description

Created a concept entry for Drag and Drop feature in SwiftUI

### Type of Change

- Adding a new entry

### CODE REVIEW

File: content/swiftui/concepts/drag-and-drop/drag-and-drop.md
Comment (SSwiniarski) [inline content/swiftui/concepts/drag-and-drop/drag-and-drop.md:2]:
```suggestion
Title: 'Drag and Drop' # Required; the file name should be the same as the title, but lowercase, with dashes instead of spaces, and all punctuation removed
```

File: content/swiftui/concepts/drag-and-drop/drag-and-drop.md
Comment (SSwiniarski) [inline content/swiftui/concepts/drag-and-drop/drag-and-drop.md:18]:
```suggestion
**`DropDelegate`** is a `protocol` in SwiftUI that lets you work with a drag and drop operations in a flexible way. To enable a view to accept drops, use the `onDrop()` method and specify a drop delegate.
```

File: content/swiftui/concepts/drag-and-drop/drag-and-drop.md
Comment (SSwiniarski) [inline content/swiftui/concepts/drag-and-drop/drag-and-drop.md:44]:
```suggestion

## DropProposal
```

File: content/swiftui/concepts/drag-and-drop/drag-and-drop.md
Comment (SSwiniarski) [inline content/swiftui/concepts/drag-and-drop/drag-and-drop.md:47]:
```suggestion
The `DropProposal` can be customized using the operation property to specify the drop operation (e.g., `.move` or `.copy`).
```

File: content/swiftui/concepts/drag-and-drop/drag-and-drop.md
Comment (SSwiniarski) [inline content/swiftui/concepts/drag-and-drop/drag-and-drop.md:46]:
```suggestion
**`DropProposal`** is a `structure` in SwiftUI that helps to control how object and drop operations should be handled when it's dropped into a SwiftUI view. It is used together with the `onDrop()` modifier to determine what happens when a user drops an object.
```

File: content/swiftui/concepts/drag-and-drop/drag-and-drop.md
Comment (SSwiniarski) [inline content/swiftui/concepts/drag-and-drop/drag-and-drop.md:96]:
```suggestion
**`DropInfo`** is a `structure` that gives information about a drop action. It is commonly used when handling drops with the `onDrop()` modifier. `DropInfo` gives details about the drop location, the types of data being dropped, and other important information.
```

File: content/swiftui/concepts/drag-and-drop/drag-and-drop.md
Comment (SSwiniarski) [inline content/swiftui/concepts/drag-and-drop/drag-and-drop.md:155]:
```suggestion
```
Delete this.

File: content/swiftui/concepts/drag-and-drop/drag-and-drop.md
Comment (SSwiniarski) [inline content/swiftui/concepts/drag-and-drop/drag-and-drop.md:99]: Please add a sentence or two introducing the example and summarizing what it is doing.

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
@arirays I did a first review and left you a few comments.

File: content/swiftui/concepts/drag-and-drop/drag-and-drop.md
Comment (SSwiniarski) [inline content/swiftui/concepts/drag-and-drop/drag-and-drop.md:18]:
```suggestion
**`DropDelegate`** is a `protocol` in SwiftUI that lets the developer work with a drag and drop operations in a flexible way. To enable a view to accept drops, use the `.onDrop()` method and specify a drop delegate.
```

File: content/swiftui/concepts/drag-and-drop/drag-and-drop.md
Comment (SSwiniarski) [inline content/swiftui/concepts/drag-and-drop/drag-and-drop.md:46]:
```suggestion
**`DropProposal`** is a structure in SwiftUI that helps to control how object and drop operations should be handled when it's dropped into a SwiftUI view. It is used together with the `.onDrop()` modifier to determine what happens when a user drops an object.
```

File: content/swiftui/concepts/drag-and-drop/drag-and-drop.md
Comment (SSwiniarski) [inline content/swiftui/concepts/drag-and-drop/drag-and-drop.md:96]:
```suggestion
**`DropInfo`** is a structure that gives information about a drop action. It is commonly used when handling drops with the `.onDrop()` modifier. `DropInfo` gives details about the drop location, the types of data being dropped, and other important information.
```

File: content/swiftui/concepts/drag-and-drop/drag-and-drop.md
Comment (SSwiniarski) [inline content/swiftui/concepts/drag-and-drop/drag-and-drop.md:100]:
```suggestion
Here is the example of a simple LazyVGrid with 4 items. We have a `.onDrag()` view modifier for each item which allows us to handle the dragging. And `.onDrop()` view modifier allows us to move and reorder elements, with the result being stored in a State variable named `draggedItem`.
```

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
@arirays A few more comments.

**Comment by arirays** [comment]
@SSwiniarski thank you for your amazing suggestions!

**Review by SSwiniarski — APPROVED** [review]
@arirays Looks good for a second review.

**Review by yangc95 — COMMENTED** [review]
@arirays Thanks for creating this one! However I noticed that we're covering how to move items using item providers. The preferred way is by moving transferable items (transferable protocol) using draggable and dropDestination.

Don't need to remove the older method (.onDrop), though it would be good to mention the difference between the two ways of creating drag and drop items

**Comment by arirays** [comment]
Hi @yangc95, I've updated drag and drop for iOS 16 with moving transferable items. 
Please let me know of any addition needed. Thanks!

File: content/swiftui/concepts/drag-and-drop/drag-and-drop.md
Comment (yangc95) [inline content/swiftui/concepts/drag-and-drop/drag-and-drop.md:17]:
```suggestion

```

File: content/swiftui/concepts/drag-and-drop/drag-and-drop.md
Comment (yangc95) [inline content/swiftui/concepts/drag-and-drop/drag-and-drop.md:18]:
```suggestion
## Move Transferable Items
```

File: content/swiftui/concepts/drag-and-drop/drag-and-drop.md
Comment (yangc95) [inline content/swiftui/concepts/drag-and-drop/drag-and-drop.md:23]:
```suggestion
Enable dragging functionality by attaching the `.draggable()` view modifier to the view and passing in the object. The object has to conform to the `Transferable` protocol.
```

File: content/swiftui/concepts/drag-and-drop/drag-and-drop.md
Comment (yangc95) [inline content/swiftui/concepts/drag-and-drop/drag-and-drop.md:22]:
```suggestion
### Dragging

```

File: content/swiftui/concepts/drag-and-drop/drag-and-drop.md
Comment (yangc95) [inline content/swiftui/concepts/drag-and-drop/drag-and-drop.md:51]:
```suggestion
### Dropping
```

File: content/swiftui/concepts/drag-and-drop/drag-and-drop.md
Comment (yangc95) [inline content/swiftui/concepts/drag-and-drop/drag-and-drop.md:52]:
```suggestion
In order to handle the drop, use `.dropDestination()`. It has two parameter values:

- `items` is an array containing all the dragged items. However, when only one item is being dragged, this array will contain just one element.
- `location` indicates the position of the dragged item within the drop view.
```

File: content/swiftui/concepts/drag-and-drop/drag-and-drop.md
Comment (yangc95) [inline content/swiftui/concepts/drag-and-drop/drag-and-drop.md:92]:
```suggestion
## Moving Items Using Item Providers 
```

File: content/swiftui/concepts/drag-and-drop/drag-and-drop.md
Comment (yangc95) [inline content/swiftui/concepts/drag-and-drop/drag-and-drop.md:24]:
```suggestion
For example, this code displays four strings representing Apple products, and allows the user to drag any of them into the box positioned below:
```

File: content/swiftui/concepts/drag-and-drop/drag-and-drop.md
Comment (yangc95) [inline content/swiftui/concepts/drag-and-drop/drag-and-drop.md:177]:
```suggestion
Here is the example of a simple LazyVGrid with 4 items. The `.onDrag()` view modifier is called on each item which handles the dragging. And the `.onDrop()` view modifier allows us to move and reorder elements, with the result being stored in a state variable named `draggedItem`.
```

**Review by yangc95 — APPROVED** [review]
This looks good to merge after resolving some formatting and wording, thanks for those changes and additions with transferable items!

File: content/swiftui/concepts/drag-and-drop/drag-and-drop.md
Comment (yangc95) [inline content/swiftui/concepts/drag-and-drop/drag-and-drop.md:8]:
```suggestion
```

**Review by yangc95 — APPROVED** [review]

File: content/swiftui/concepts/drag-and-drop/drag-and-drop.md
Comment (yangc95) [inline content/swiftui/concepts/drag-and-drop/drag-and-drop.md:16]:
```suggestion

## Move Transferable Items
```

File: content/swiftui/concepts/drag-and-drop/drag-and-drop.md
Comment (yangc95) [inline content/swiftui/concepts/drag-and-drop/drag-and-drop.md:50]:
```suggestion
### Dropping

```

File: content/swiftui/concepts/drag-and-drop/drag-and-drop.md
Comment (yangc95) [inline content/swiftui/concepts/drag-and-drop/drag-and-drop.md:123]:
```suggestion
## `DropProposal`
```

File: content/swiftui/concepts/drag-and-drop/drag-and-drop.md
Comment (yangc95) [inline content/swiftui/concepts/drag-and-drop/drag-and-drop.md:126]:
```suggestion

`DropProposal` can be customized using the operation property to specify the drop operation (e.g., `.move` or `.copy`).
```

File: content/swiftui/concepts/drag-and-drop/drag-and-drop.md
Comment (yangc95) [inline content/swiftui/concepts/drag-and-drop/drag-and-drop.md:173]:
```suggestion
## DropInfo
```

File: content/swiftui/concepts/drag-and-drop/drag-and-drop.md
Comment (yangc95) [inline content/swiftui/concepts/drag-and-drop/drag-and-drop.md:177]:
```suggestion
## Example Using `.onDrag()` and `.onDrop()`
```

**Pull request merged**
